sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -696,6 +713,90 @@ class CudfRegexTranspiler(mode: RegexMode) {
     }
   }
 
+  private def transpile(regex: RegexAST, replacement: Option[RegexReplacement],
+      previous: Option[RegexAST]): RegexAST = {
```

Comment:
nit: `replacement` and `previous` is never used here, just passed on to `rewrite`. It may be more descriptive to remove these, rename this method to `checkUnsupported` (take the code out of the closure), and then call this and `rewrite` inside `transpile`